### PR TITLE
compute: add JSON explain output for `bin/mzexplore`

### PR DESCRIPTION
**Pull request by mgree**
`bin/mzexplore` can extract `EXPLAIN PLAN` output as text.

This PR adds a flag `--format [FORMAT]` that lets you get JSON output.

Along the way, there are a few minor bugfixes.

### Motivation

  * This PR fixes a previously unreported bug.

See the first two commits: there was a copypasta mistake in docs; the help text for `bin/mzexplore extract --help` gave incorrect output for `plans`. The third commit and also fixes an issue that led to `..txt` suffixes in generated plans.

  * This PR adds a feature that has not yet been specified.

Running `bin/mzexplore extract plans --format json ...` will extract plans in JSON format.

### Tips for reviewer

First two commits are tiny bugfixes; third commit is the new behavior.

### Checklist

- [ ] This PR has adequate test coverage / QA involvement has been duly considered.
- [ ] This PR has an associated up-to-date [design doc](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/README.md), is a design doc ([template](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/00000000_template.md)), or is sufficiently small to not require a design.
  
- [ ] If this PR evolves [an existing `$T ⇔ Proto$T` mapping](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/command-and-response-binary-encoding.md) (possibly in a backwards-incompatible way), then it is tagged with a `T-proto` label.
- [ ] If this PR will require changes to cloud orchestration or tests, there is a companion cloud PR to account for those changes that is tagged with the release-blocker label ([example](https://github.com/MaterializeInc/cloud/pull/5021)).
  
- [ ] This PR includes the following [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/guide-changes.md#what-changes-require-a-release-note):
  -

### CODE REVIEW

File: bin/mzexplore [hunk 1]
```diff
@@ -9,6 +9,6 @@
 # the Business Source License, use of this software will be governed
 # by the Apache License, Version 2.0.
 #
-# dashboard-clean - Clean the dashboard model json to be fit for public consumption
+# mzexplore - Extract definitions and plans from the cloud
```
Comment (aalexandrov) [inline bin/mzexplore:12]:
```suggestion
# mzexplore - Extract definitions and plans from a Materialize instance using pgwire.
```

File: misc/python/materialize/cli/mzexplore.py [hunk 2]
```diff
@@ -178,8 +186,9 @@ def defs(
 @click.option("--explainee-type", "-t", **Opt.explainee_type)
 @click.option("--with", "-w", "explain_flags", **Opt.explain_flags)
 @click.option("--stage", "-s", "explain_stages", **Opt.explain_stage)
+@click.option("--format", "-f", "explain_format", **Opt.explain_format)
 @click.option("--suffix", **Opt.explain_suffix)
-@is_documented_by(api.extract.defs)
+@is_documented_by(api.extract.plans)
```
Comment (aalexandrov) [inline misc/python/materialize/cli/mzexplore.py:191]: Ooops...

**Review by aalexandrov — APPROVED** [review]

File: misc/python/materialize/mzexplore/extract.py [hunk 3]
```diff
@@ -233,10 +236,7 @@ def plans(
                     info(f"Explaining {stage} for CREATE {fqname} in `{file_name}`")
                     try:
                         plans[file_name] = explain(
-                            db,
-                            stage,
-                            explainee,
-                            explain_flags,
+                            db, stage, explainee, explain_flags, explain_format
```
[the remark this answers is not in the document]
Comment (mgree) [reply misc/python/materialize/mzexplore/extract.py]: Done!

File: misc/python/materialize/mzexplore/extract.py [hunk 4]
```diff
@@ -183,9 +185,9 @@ def plans(
                                 ExplaineeType.CATALOG_ITEM,
                                 stage.name.lower(),
                                 suffix,
-                                "txt",
+                                explain_format.suffix(),
                             ]
-                            if part is not None
```
[the remark this answers is not in the document]
Comment (aalexandrov) [reply misc/python/materialize/mzexplore/extract.py:188]: I'll rebase on top of this if it lands first.

**Comment by mgree** [comment]
Ugh, some misstep with the rebase broke the history. Trying again in #25695.

**Pull request closed without merging**